### PR TITLE
chipsalliance/Surelog#2541 Access to included files

### PR DESCRIPTION
**Pull request by hs-apotell**
* Fixing a number of location issues
* Updating IncludeFileInfo to include startColumn, endLine and endColumn
* Adding access to IncludeFileInfo without opening up the guts of the library
* Fixed a number of warnings
* Few minor optimizations

### CODE REVIEW

**Review by hzeller — COMMENTED** [review]
Mostly LGTM w.r.t. some cleanup changes, possibly something for future PRs (I'll let Alain have the final word, also about opening up the exported internal headers, which might result in some future headache to maintain)

File: scripts/regression.py [hunk 1]
```diff
@@ -29,6 +29,7 @@
 # either to the workspace directory or the build directory
 _default_test_dirpaths = [ 'tests', os.path.join('third_party', 'tests') ]
 _default_build_dirpath = 'build'
+# _default_build_dirpath = os.path.join('out', 'build', 'Debug')
```
[the remark this answers is not in the document]
Comment (hs-apotell) [reply scripts/regression.py:32]: This PR is not ready to be merged by any means. It's still work in progress. This was just part of my debugging on Windows.
This changes is likely to be split into at least a few PRs.

Posted to get help from @alaindargelas on include file info location. 3 of the 4 components of the information is missing.

File: src/SourceCompile/IncludeFileInfo.h [hunk 2]
```diff
@@ -34,40 +34,61 @@ class IncludeFileInfo {
...
+        m_originalStartColumn(i.m_originalStartColumn),
+        m_originalEndLine(i.m_originalEndLine),
+        m_originalEndColumn(i.m_originalEndColumn),
         m_type(i.m_type),
         m_indexOpening(i.m_indexOpening),
         m_indexClosing(i.m_indexClosing) {}
   IncludeFileInfo(unsigned int sectionStartLine, SymbolId sectionFile,
-                  unsigned int originalLine, Action type, int indexOpening,
-                  int indexClosing)
+                  unsigned int originalStartLine,
```
[the remark this answers is not in the document]
Comment (hs-apotell) [reply src/SourceCompile/IncludeFileInfo.h:69]: I agree. Position and PositionRange is what I had in mind but that's a rather larger change. The focus at the moment was to get the correct information filled in so we can continue our work on the LS module. There is lot of missing information in this context.

**Comment by hs-apotell** [comment]
> about opening up the exported internal headers

Temporary change made to keep moving while we narrowed down on a longer term solution. The solution we discussed was to push the IncludeFileInfo like structure in the UHDM::design model. This would avoid opening the internals of the parsing and compiling to public API.

**Comment by alaindargelas** [comment]
Looks OK in principle, as you said, please break it down in PRs of relevant changes.
I'll fill in the column info in the AnalyzeFile after this is pushed, please remind me after you push the relevant PR.
Basically, AnalyzeFile is a character based fast parser, one has to monitor carriage return to know which column you are in.
That should be a simple change.

**Comment by hs-apotell** [comment]
Opened a narrowed down PR specific to this issue - https://github.com/chipsalliance/Surelog/pull/2586

**Pull request closed without merging**